gpkit/constraints/sgp.py

```diff
@@ -45,37 +45,34 @@ def __init__(self, cost, constraints, substitutions=None, verbosity=1):
         self.gps = []
         self.results = []
         self.result = None
-        self.lastgp = None
-        self.is_sgp = False
         self._spconstrs = []
         self._approx_lt = []
-        self._gppos = None
+        self._numgpconstrs = None
+        self._gp = None
 
         if cost.any_nonpositive_cs:
-            raise TypeError("""SignomialPrograms need Posyomial objectives.
+            raise TypeError("""Sequential GPs need Posynomial objectives.
 
     The equivalent of a Signomial objective can be constructed by constraining
-    a dummy variable z to be greater than the desired Signomial objective s
+    a dummy variable `z` to be greater than the desired Signomial objective `s`
     (z >= s) and then minimizing that dummy variable.""")
         CostedConstraintSet.__init__(self, cost, constraints, substitutions)
-        try:
-            self.__parse_externalfnvars()
-            if self.externalfn_vars:  # not a GP! Skip to the `except`
-                raise InvalidGPConstraint("some variables have externalfns")
-            _ = self.as_posyslt1(substitutions)
-        except InvalidGPConstraint:
-            pass
-        else:  # this is a GP
-            raise ValueError("""Model valid as a Geometric Program.
-
-    SignomialPrograms should only be created with Models containing Signomial
+        self.externalfn_vars = frozenset(Variable(newvariable=False, **v.descr)
+                                         for v in self.varkeys if v.externalfn)
+        self.not_sp = bool(self.externalfn_vars)
```

Comment:
(note is_sgp/not_sp are not characterstics of this refactor, but rather of the FastSP refactor from may)